components/Input/MultiSelectInput/index.js

```diff
@@ -340,9 +342,78 @@ class MultiSelectInput extends React.PureComponent {
             displayOptions,
             inputValue: value,
             showOptions: true,
+            focusedKey: undefined,
         });
     }
 
+    handleInputKeyDown = (e) => {
+        const { focusedKey, displayOptions, showOptions } = this.state;
+        const { keySelector } = this.props;
+
+        // Keycodes:
+        // 9: Tab
+        // 27: Escape
+        // 13: Enter
+        // 32: Space
+        // 38: Down
+        // 40: Up
+
+        // If tab or escape was pressed and dropdown is being shown,
+        // hide the dropdown.
+        if ((e.keyCode === 9 || e.keyCode === 27) && showOptions) {
+            this.toggleDropdown();
+            return;
+        }
+
+        // List of special keys, we are going to handle
+        const specialKeys = [40, 38, 13, 32];
+        if (displayOptions.length === 0 || specialKeys.indexOf(e.keyCode) === -1) {
+            return;
+        }
+
+        // Handle space and enter keys only if an option is focused
+        if ((e.keyCode === 13 || e.keyCode === 32) && !focusedKey) {
+            return;
+        }
+
+        // First, disable default behaviour for these keys
+        e.stopPropagation();
+        e.preventDefault();
+
+        // If any of the special keys was pressed but the dropdown is currently hidden,
+        // show the dropdown.
+        if (!showOptions) {
+            this.toggleDropdown();
+            return;
+        }
+
+        if (e.keyCode === 13 || e.keyCode === 32) {
+            e.stopPropagation();
```

Comment:
e.stopPropagation and e.preventDefault was already invoked once before this point.